BOARD: Enable PCA6408 on FRDM-IMX93 board

CkovMk opened the pull request:
- [x] depends on #93242

CkovMk commented:
Preserving pca_series commits for CI check. Once CI passes I will drop commits from the other PR and make it clean for review.

CkovMk commented:
re-trigger ci

CkovMk commented:
Rebase to main

JiafeiPan commented on boards/nxp/frdm_imx93/frdm_imx93_mimx9352_a55.dts:
```diff
@@ -88,11 +88,34 @@
+		 * the same init level. Must do deferred initialization after reset pin
+		 * is released.
+		 */
+		zephyr,deferred-init;
```
As I know it will automatically set initialization sequence according to the dependency as exp0 and exp1 have same init priority but exp1 depends on exp0, so exp0 should be initialized firstly, then exp1 will be initialized, so no need to use deffered-init

CkovMk committed:
board: frdm_imx93: enable pcal6408 gpio expander

Enable  pcal6408 gpio expander on FRDM-IMX93 board. This expander has
its reset pin controlled by another expander, and both expanders is
driven by the same pca_series driver. To guarentee correct init order,
pcal6408 is set to deferred init in a board init function.

Signed-off-by: Chekhov Ma <chekhov.ma@nxp.com>
